lib/susedistribution.pm

```diff
@@ -571,10 +571,33 @@ sub init_consoles {
             my $packed_ip = gethostbyname($s390_guest_fqdn);
             die "Failed to get host by name for '$s390_guest_fqdn' (on " . hostname . ")" unless $packed_ip;
             my $s390_guest_ip = inet_ntoa($packed_ip);
-            my $host_ip = get_var("AGAMA") ? 'hostip' : 'HostIP';
-            my $host_name = get_var("AGAMA") ? 'hostname' : 'Hostname';
-            $s390_params .= " $host_ip" . "=${s390_guest_ip}/${s390_guest_subnetmask}";
-            $s390_params .= " $host_name" . "=${s390_guest_hostname}";
+            if(get_var("AGAMA")) {
+                my @split = split(/\./, get_required_var("REPO_HOST"));
+                my $s390_gateway_ip = join(".", @split[0], @split[1], @split[2], "254");
+
+                my ($hostname) = split(/\./, get_required_var("ZVM_GUEST"));
+
+                if(is_sle()){
+                    @split = split(/\./, get_required_var("ZVM_GUEST"));
+                    my $s390_domain = join( ".", @split[1], @split[2], @split[3], @split[4]);
+
+                    my $s390_device = "enc800";
+
+                    $s390_params .= " ip=${s390_guest_ip}:${s390_guest_hostname}:${s390_gateway_ip}:${s390_guest_subnetmask}:${s390_domain}:${s390_device}:none";
+                }
+                else {
+                    @split = split(/\./, get_required_var("ZVM_GUEST"));
+                    my $s390_domain = join( ".", @split[1], @split[2], @split[3]);
+
+                    my $s390_device = "enca00";
+
+                    $s390_params .= " ip=${s390_guest_ip}:${s390_guest_hostname}:${s390_gateway_ip}:${s390_guest_subnetmask}:${s390_domain}:${s390_device}:none";
```

Comment:
we are repeating code here, we could just run this once and get the values depending on the case before. Same for other lines that could be similar.

Reply:
6742aaafb